posts/2023-08-22-ArgoCD-Configuration-Drift.adoc

```diff
@@ -0,0 +1,156 @@
+
+To begin, navigate to GitHub and fork this repository: https://github.com/WASdev/sample.daytrader7. Use your forked repository when configuring ArgoCD in the following steps
+
+We assume that your CI pipleline has a way to build contianer image for your application. For this example, we have a pre-built cotnainer image used in the deployment files `deploy/daytrader7-deploy.yaml`.
+
+== Deploying DayTrader 7 via GitOps ==
+
+Now that we have the Git configuration repository ready for deployment, it’s time to configure ArgoCD to deploy the application.
+
+Go to the Argo CD console and ensure that you are logged in.
+Click the New App button near the top left of the console.
+For Application name, let’s call this one daytrader7, keeping it in the default Argo project and setting the sync policy to Manual. Check the Auto-Create Namespace box as well.
+[.img_border_light]
+image::/img/blog/configuration-drift-3.png[Creating the App,width=70%,align="center"]
+Scroll down to the Source and change the Repository URL to your newly forked repo. Change the path, which is the path where Argo CD is looking for deployment files, to deploy. Set the Cluster URL to https://kubernetes.default.svc, which is the local cluster URL. Then set the namespace to daytrader7
+Click Create in the top left.
+[.img_border_light]
+image::/img/blog/configuration-drift-4.png[Setting the Repo,width=70%,align="center"]
+
+== Syncing DayTrader ==
+On the main page of the Argo CD console, you should see a new tile that looks like this:
+[.img_border_light]
+image::/img/blog/configuration-drift-5.png[Argo Dashboard,width=70%,align="center"]
+Click the Sync button, then on the window that appears and click Sync again.
+Click on the daytrader7 tile to view the app dashboard and sync progress.
+Over the course of a few minutes, you should see resources being created and the app dashboard looking like this:
+[.img_border_light]
+image::/img/blog/configuration-drift-6.png[App deploying,width=70%,align="center"]
+Now the app is deployed and can be reached in a route created under the daytrader7 namespace. Find the URL via oc get routes –n daytrader7, then paste the URL in a browser.
+And there we go! The app is fully deployed and ready to be used.
+[.img_border_light]
+image::/img/blog/configuration-drift-7.png[Daytrader Home,width=70%,align="center"]
+== Demo Configuration Drift ==
+Now that we have the app deployed let's show a scenario where configuration drift occurs and how we can easily correct it.  Let's say a developer changes the memory settings in the configuration file for a quick test.  With the Kubernetes CLI or console change the memory requests from `1024Mi` to `2048Mi` if this resource: `deploy/daytrader7-deploy.yaml`.  In a few moments the argo console should show it is out of sync:
```

Comment:
I put put the "let's say..." and below as a new para

Reply:
should it be "in" this resource instead of if?

Reply:
I would add "To represent this, change the memory request for the deploy/daytrader7-deploy.yaml file from `1024Mi` to `2048Mi` using either the Kubernetes CLI or the console itself." This way it links to the sentence before more and I understand why I'm doing it before the instruction for how to do it.

Reply:
why is it showing it's out of sync significant? detail this here with a short sentence.